rust/operator-binary/src/superset_db_controller.rs

```diff
@@ -0,0 +1,235 @@
+    InitStatusMissing { namespace: String, name: String },
+    #[snafu(display("object is missing metadata to build owner reference"))]
+    ObjectMissingMetadataForOwnerRef {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display(
+        "database state is 'initializing' but failed to find job {}/{}",
+        namespace,
+        name
+    ))]
+    GetInitializationJob {
+        source: stackable_operator::error::Error,
+        namespace: String,
+        name: String,
+    },
+    #[snafu(display("Failed to check whether the secret ({}) exists", secret_name))]
+    SecretCheck {
+        source: stackable_operator::error::Error,
+        secret_name: String,
+    },
+}
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub async fn reconcile_superset_db(
+    superset_db: SupersetDB,
+    ctx: Context<Ctx>,
+) -> Result<ReconcilerAction> {
+    tracing::info!("Starting reconcile");
+
+    let client = &ctx.get_ref().client;
+
+    if let Some(ref s) = superset_db.status {
+        match s.condition {
+            SupersetDBStatusCondition::Provisioned => {
+                let secret_exists = client
+                    .exists::<Secret>(
+                        &superset_db.spec.credentials_secret,
+                        superset_db.metadata.namespace.as_deref(),
+                    )
+                    .await
+                    .context(SecretCheck {
+                        secret_name: superset_db.spec.credentials_secret.clone(),
+                    })?;
+                if secret_exists {
+                    let job = build_init_job(&superset_db)?;
+                    client
+                        .apply_patch(FIELD_MANAGER_SCOPE, &job, &job)
+                        .await
+                        .context(ApplyJob {
+                            superset_db: ObjectRef::from_obj(&superset_db),
+                        })?;
+                    // The job is started, update status to reflect new state
+                    client
+                        .apply_patch_status(FIELD_MANAGER_SCOPE, &superset_db, &s.initializing())
+                        .await
+                        .context(ApplyStatus)?;
+                }
+            }
+            SupersetDBStatusCondition::Initializing => {
+                // In here, check the associated job that is running.
+                // If it is still running, do nothing. If it completed, set status to ready, if it failed, set status to failed.
+                // TODO we need to fetch the job here
+                // we need namespace/name.
+                let ns = superset_db.metadata.namespace.clone().unwrap();
+                let job_name = superset_db.job_name();
+                let job = client.get::<Job>(&job_name, Some(&ns)).await.context(
+                    GetInitializationJob {
+                        namespace: ns,
+                        name: job_name,
+                    },
+                )?;
```

Comment:
name(), namespace(), ObjectRef<_> -> check druid_connection_controller

Reply:
I did it